Source/WebCore/rendering/RenderFlexibleBox.cpp

```diff
@@ -2221,31 +2226,27 @@ void RenderFlexibleBox::alignChildren(const Vector<LineContext>& lineContexts)
                 continue;
             
             ItemPosition position = alignmentForChild(flexItem.box);
+            LayoutUnit maxAscent = position == ItemPosition::LastBaseline ? lineContext.lastBaselineMaxAscent : lineContext.maxAscent;
             if (position == ItemPosition::Stretch)
                 applyStretchAlignmentToChild(flexItem.box, lineCrossAxisExtent);
             LayoutUnit availableSpace = availableAlignmentSpaceForChild(lineCrossAxisExtent, flexItem.box);
             LayoutUnit offset = alignmentOffset(availableSpace, position, marginBoxAscentForChild(flexItem.box), maxAscent, style().flexWrap() == FlexWrap::Reverse);
             adjustAlignmentForChild(flexItem.box, offset);
-            if (position == ItemPosition::Baseline && style().flexWrap() == FlexWrap::Reverse)
+            if (shouldAdjustItemTowardsCrossAxis(flexItem.box, position)) 
                 minMarginAfterBaseline = std::min(minMarginAfterBaseline, availableAlignmentSpaceForChild(lineCrossAxisExtent, flexItem.box) - offset);
         }
 
-        minMarginAfterBaselines.append(minMarginAfterBaseline);
-    }
-    
-    if (style().flexWrap() != FlexWrap::Reverse)
-        return;
-    
-    // wrap-reverse flips the cross axis start and end. For baseline alignment,
-    // this means we need to align the after edge of baseline elements with the
-    // after edge of the flex line.
-    for (size_t lineNumber = 0; lineNumber < lineContexts.size(); ++lineNumber) {
-        const LineContext& lineContext = lineContexts[lineNumber];
-        LayoutUnit minMarginAfterBaseline = minMarginAfterBaselines[lineNumber];
-        for (size_t childNumber = 0; childNumber < lineContext.flexItems.size(); ++childNumber) {
-            const auto& flexItem = lineContext.flexItems[childNumber];
-            if (alignmentForChild(flexItem.box) == ItemPosition::Baseline && !hasAutoMarginsInCrossAxis(flexItem.box) && minMarginAfterBaseline)
-                adjustAlignmentForChild(flexItem.box, minMarginAfterBaseline);
+        // wrap-reverse flips the cross axis start and end. For baseline alignment,
+        // this means we need to align the after edge of baseline elements with the
+        // after edge of the flex line. We can also use the same logic for last
+        // baseline alignment since those items must be pushed towards the cross
+        // start as well.
+        if (minMarginAfterBaseline) {
+            for (size_t childNumber = 0; childNumber < lineContext.flexItems.size(); ++childNumber) {
```

Comment:
I think this could be a modern C++ for loop:

```
for (auto& flexItem : lineContext.flexItems) {
  if (shouldAdjustItemTowardsCrossAxis(flexItem.box, alignmentForChild(flexItem.box)) && !hasAutoMarginsInCrossAxis(flexItem.box))
                     adjustAlignmentForChild(flexItem.box, minMarginAfterBaseline);
}
```

Reply:
Fixed! There were also a couple of other places above in the same method where I made the same change.